src/layouts/Catalogue/Catalogue.js

```diff
@@ -1,358 +1,149 @@
+  font-size: 50pt;
+  font-weight: 700;
+  text-align: center;
+`;
+
+const Products = styled.ul`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-direction: row;
+  flex-wrap: wrap;
+  width: 75%;
+`;
+
+export default function Catalogue() {
+  const [products, setProducts] = useState([]);
+  const [showModalCart, setShowModalCart] = useState(false);
+
+  const { state, actions } = useProductReducer();
+  const { stateCart, actionCarts } = useCartReducer();
+
+  useEffect(() => {
+    ProductApi.get()
+      .then((res) => res.json())
+      .then((res) => {
+        console.log(`res`, res);
+        setProducts(res);
+      });
+  }, []);
+
+  function showModal(product) {
+    actions.changeValue(product);
   }
 
-  getDataProducts() {
-    return ProductApi.get().then(res => res.json())
-    .then(res => {
-      this.setState({
-        products : res,
-      })
-    });
+  function handlerEmptyCart() {
+    actionCarts.reset();
   }
 
-  componentDidMount() {
-    document.title = "Katalog Produk";
-    this.getDataProducts();
+  function handleCancel() {
+    actions.reset();
   }
 
-  showModal(product){
-    this.setState({showModal: true, selectedProduct: product})
+  function handleCancelCart() {
+    setShowModalCart(false);
   }
 
-  handleOk = () => {
-    this.setState({ loading: true });
-    setTimeout(() => {
-      this.setState({ loading: false, showModal: false, selectedProduct: {} , qty: 1});
-    }, 3000);
-  };
-
-  handleCancel = () => {
-    this.setState({ showModal: false, selectedProduct: {}, qty: 1 });
-  };
-
-  handleOkCart = () => {
-    this.setState({ showModalCart: false });
-  };
-
-  handleCancelCart = () => {
-    this.setState({ showModalCart: false });
-  };
-
-  addtoCartHandler = () => {
-    var carts = this.state.cart
-    var qty = this.state.qty
+  function addtoCartHandler() {
+    var carts = stateCart.cart;
+    var qty = state.quantity;
     var position = carts
-    .map(function (e) {
-      return e.id;
-    })
-    .indexOf(this.state.selectedProduct.id);
-    if (position<0){
+      .map(function (e) {
+        return e.id;
+      })
+      .indexOf(state.selectedProduct.id);
+    if (position < 0) {
       //new insert
       var objInserted = {
-        ...this.state.selectedProduct,
+        ...state.selectedProduct,
         qty,
       };
-      carts.push(objInserted)
-    }else{
+      carts.push(objInserted);
+    } else {
       var obj = carts[position];
-      var objInserted = {...obj, qty: obj.qty+this.state.qty};
+      var objInserted = { ...obj, qty: obj.qty + state.quantity };
       carts[position] = objInserted;
     }
-    this.setState({
-      showModal: false,
-      qty: 1,
-      selectedProduct: {},
-    },()=>{
-      this.recalculateGrandTotal();
-      console.log(`cart`, this.state.cart)
-    })
-  }
-
-  handlerEmptyCart(){
-    this.setState({
-      cart: [],
-    })
+    actions.reset();
+    recalculateGrandTotal();
```

Comment:
I think this functionality should be handled by a reducer as well. This looks a bit fuzzy.

It's unclear who's responsible for keeping the state of the cart. I think it should be the `cartReducer`, but now you modify the state directly in this line `carts[position] = objInserted;`, which is a bad practice. You should always keep your state in `useState` or `useReducer` (or more advanced solutions like Context).